src/werkzeug/formparser.py

```diff
@@ -560,7 +433,12 @@ def parse_parts(self, file, boundary, content_length):
                     part_charset = self.get_part_charset(headers)
                     yield (
                         "form",
-                        (name, b"".join(container).decode(part_charset, self.errors)),
+                        (
+                            name,
+                            self.build_field_storage(
+                                name, container, part_charset, headers
```

Comment:
This doesn't match the API of `FileStorage`, where the parameter order is `container, name, headers`. Maybe call `get_part_charset` in `build_field_storage` instead.

Reply:
Happy to align argument order to that of the `FileStorage` constructor. Also agree that it makes sense for `get_part_charset` to go into `build_field_storage` (I didn't originally as I tried to make as few changes as possible). I will update shortly.

Reply:
Suggestions implemented in 557c44f19785422b4a948f183654218604558c97, 8a0c7b59b15a68ec948a122be9dd99ee6ca80877.